base_layer/core/src/base_node/proto/request.rs

```diff
@@ -86,6 +112,27 @@ impl From<ci::NodeCommsRequest> for ProtoNodeCommsRequest {
             FetchTxos(hash_outputs) => ProtoNodeCommsRequest::FetchTxos(hash_outputs.into()),
             FetchBlocks(block_heights) => ProtoNodeCommsRequest::FetchBlocks(block_heights.into()),
             FetchBlocksWithHashes(block_hashes) => ProtoNodeCommsRequest::FetchBlocksWithHashes(block_hashes.into()),
+            FetchBlocksWithKernels(signatures) => {
+                let mut sigs = Vec::new();
+                for sig in signatures {
+                    sigs.push(sig.into())
+                }
```

Comment:
nit:
```suggestion
                let sigs = signatures.into_iter().map(Into::into).collect();
```

Same for below. This will also only allocate `sigs` once (Vec::with_capacity)